framework/src/transfers/MultiAppMeshFunctionTransfer.C

```diff
@@ -63,289 +62,347 @@ MultiAppMeshFunctionTransfer::execute()
+              point_found = true;
             }
           }
-          else // Elemental
-          {
-            MeshBase::const_element_iterator elem_it = mesh.local_elements_begin();
-            MeshBase::const_element_iterator elem_end = mesh.local_elements_end();
+        }
+      }
+    }
+    else // Elemental
+    {
+      MeshBase::const_element_iterator elem_it = to_mesh->local_elements_begin();
+      MeshBase::const_element_iterator elem_end = to_mesh->local_elements_end();
+
+      for (; elem_it != elem_end; ++elem_it)
+      {
+        Elem * elem = *elem_it;
+
+        Point centroid = elem->centroid();
+
+        // Skip this element if the variable has no dofs at it.
+        if (elem->n_dofs(sys_num, var_num) < 1)
+          continue;
 
-            for (; elem_it != elem_end; ++elem_it)
+        // Loop over the "froms" on processor i_proc.  If the elem is found in
+        // any of the "froms", add that elem to the vector that will be sent to
+        // i_proc.
+        unsigned int from0 = 0;
+        for (processor_id_type i_proc = 0;
+             i_proc < n_processors();
+             from0 += froms_per_proc[i_proc], i_proc++)
+        {
+          bool point_found = false;
+          for (unsigned int i_from = from0; i_from < from0 + froms_per_proc[i_proc] && ! point_found; i_from++)
+          {
+            if (bboxes[i_from].contains_point(centroid + _to_positions[i_to]))
             {
-              Elem * elem = *elem_it;
-
-              Point centroid = elem->centroid();
-
-              if (elem->n_dofs(sys_num, var_num) > 0) // If this variable has dofs at this elem
-              {
-                // The zero only works for LAGRANGE!
-                dof_id_type dof = elem->dof_number(sys_num, var_num, 0);
-
-                // Swap back
-                Moose::swapLibMeshComm(swapped);
-                Real from_value = from_func(centroid+_multi_app->position(i));
-                // Swap again
-                swapped = Moose::swapLibMeshComm(_multi_app->comm());
-
-                if (from_value != OutOfMeshValue)
-                  solution.set(dof, from_value);
-                else if (_error_on_miss)
-                  mooseError("Point not found! " << centroid+_multi_app->position(i) << std::endl);
-              }
+              std::pair<unsigned int, unsigned int> key(i_to, elem->id());
+              point_index_map[i_proc][key] = outgoing_points[i_proc].size();
+              outgoing_points[i_proc].push_back(centroid + _to_positions[i_to]);
+              point_found = true;
             }
           }
-
-          solution.close();
-          to_sys->update();
-
-          // Swap back
-          Moose::swapLibMeshComm(swapped);
         }
       }
+    }
+  }
 
-      delete serialized_solution;
+  ////////////////////
+  // Request point evaluations from other processors and handle requests sent to
+  // this processor.
+  ////////////////////
```

Comment:
Hm... according to [this](http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html) it looks like your format will work to define a code block in Doxygen.  I guess I'd still prefer 
```
/**
 *
 */
```
to do that, since there is precedent for it in other parts of the framework, but if that doesn't separate things enough for you it's also fine as-is.